include/smack/SmackRep.h

```diff
@@ -78,16 +79,17 @@ class SmackRep {
     : aliasAnalysis(aa), naming(N), program(P),
       targetData(aa->getDataLayout()), globalsBottom(0) {
     uniqueFpNum = 0;
+    ptrSizeInBits = targetData->getPointerSizeInBits();
   }
   DSAAliasAnalysis* getAliasAnalysis() { return aliasAnalysis; }
   Program& getProgram() { return program; }
 
 private:
-  void addInit(unsigned region, const Expr* addr, const llvm::Constant* val);
+  void addInit(unsigned region, const Expr* addr, const llvm::Constant* val, const llvm::GlobalValue* V, bool safety);
 
-  const Expr* pa(const Expr* base, int index, int size);
-  const Expr* pa(const Expr* base, const Expr* index, int size);
-  const Expr* pa(const Expr* base, const Expr* index, const Expr* size);
+  const Expr* pa(const Expr* base, int index, int size, int i_size = 0, int t_size = 0);
+  const Expr* pa(const Expr* base, const Expr* index, int size, int i_size = 0, int t_size = 0);
+  const Expr* pa(const Expr* base, const Expr* index, const Expr* size, int i_size = 0, int t_size = 0);
   
   const Expr* b2p(const llvm::Value* v);
```

Comment:
My mistake. I thought you were referring to `pa` functions.

On Fri, Feb 6, 2015, 12:47 PM Michael Emmi <notifications@github.com> wrote:

> In include/smack/SmackRep.h
> <https://github.com/smackers/smack/pull/56#discussion_r24266556>:
>
> >
> > -  const Expr* pa(const Expr* base, int index, int size);
> > -  const Expr* pa(const Expr* base, const Expr* index, int size);
> > -  const Expr* pa(const Expr* base, const Expr* index, const Expr* size);
> > +  const Expr* pa(const Expr* base, int index, int size, int i_size = 0, int t_size = 0);
> > +  const Expr* pa(const Expr* base, const Expr* index, int size, int i_size = 0, int t_size = 0);
> > +  const Expr* pa(const Expr* base, const Expr* index, const Expr* size, int i_size = 0, int t_size = 0);
> >
> >    const Expr* b2p(const llvm::Value* v);
>
> How do you figure b2p is not obsolete? The following grep shows that b2p
> is no longer used in SMACK code.
>
> grep "[Bb]2[Pp]" -r include/ lib/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/smackers/smack/pull/56/files#r24266556>.
>